Week5/MyFavoriteMovies/app/src/main/java/com/raywenderlich/myfavoritemovies/activities/LoginActivity.kt

```diff
@@ -0,0 +1,52 @@
+package com.raywenderlich.myfavoritemovies.activities
+
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import android.widget.Toast
+import androidx.appcompat.app.AppCompatActivity
+import com.raywenderlich.myfavoritemovies.R
+import kotlinx.android.synthetic.main.activity_login.*
+
+class LoginActivity : AppCompatActivity() {
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_login)
+
+        val sharedPrefs =
```

Comment:
It doesnt' make sense to fetch preferences in LoginActivity, using a key from the MainActivity.

Also, the preferences shouldn't be named according to "LOGGED_IN_STATUS_KEY", they should have a name that represents those preferences, e.g. "FavoriteMoviesPreferences".

Also, it would be better if you had reused the preferences, by creating a singleton object within the App class, as we had discussed previously, but I think you submitted the HW before we had that discussion. So make sure to follow that approach in the future.

Once again, split code up into multiple functions when working with different parts of the code. This is a really important thing in development, and I keep mentioning that you should split up your code into smaller functions, and not just have everything in `onCreate()` or similar.